### ISSUE TITLE
Excess solver queries during concrete execution

### ISSUE THREAD

**Issue by ttuegel** [runtimeverification/haskell-backend#1490]
During concrete execution, using `--smt none`, we see a large number of queries to the solver of the form of remainders:

```
kore-exec: [2020-01-30 15:38:03.814882846] Warning (LogMessage): Failed to decide predicate:
    \not{_PREDICATE{}}(
        \equals{SortEthereumCell{}, _PREDICATE{}}(
            Var'Unds'32:SortEthereumCell{},
            Lbl'-LT-'ethereum'-GT-'{}(
                Lbl'-LT-'evm'-GT-'{}(
                    Var'Unds'13:SortOutputCell{},
                    Var'Unds'14:SortStatusCodeCell{},
                    Var'Unds'15:SortCallStackCell{},
                    Var'Unds'16:SortInterimStatesCell{},
                    Var'Unds'17:SortTouchedAccountsCell{},
                    Lbl'-LT-'callState'-GT-'{}(
                        Var'Unds'0:SortProgramCell{},
                        Var'Unds'1:SortJumpDestsCell{},
                        Var'Unds'2:SortIdCell{},
                        Var'Unds'3:SortCallerCell{},
                        Var'Unds'4:SortCallDataCell{},
                        Var'Unds'5:SortCallValueCell{},
                        Lbl'-LT-'wordStack'-GT-'{}(VarWS:SortWordStack{}),
                        Var'Unds'6:SortLocalMemCell{},
                        Var'Unds'7:SortPcCell{},
                        Var'Unds'8:SortGasCell{},
                        Var'Unds'9:SortMemoryUsedCell{},
                        Var'Unds'10:SortCallGasCell{},
                        Var'Unds'11:SortStaticCell{},
                        Var'Unds'12:SortCallDepthCell{}
                    ),
                    Var'Unds'18:SortSubstateCell{},
                    Var'Unds'19:SortGasPriceCell{},
                    Var'Unds'20:SortOriginCell{},
                    Var'Unds'21:SortBlockhashesCell{},
                    Var'Unds'22:SortBlockCell{}
                ),
                VarDotVar3:SortNetworkCell{}
            )
        )
    ) []
```

Given that both sides of this equality are symbolic, I suspect this happens at startup, when we simplify rules. We shouldn't have remainders during concrete execution, so otherwise I can't imagine what is happening.

Related: #1407

**Comment by ttuegel** [comment]
My initial suggestion for this would be to avoid the solver entirely when we do start-up simplification. I think the purpose of that simplification pass is to run our _internal_ simplification routines over the rules. If I recall correctly, @virgil-serbanuta and @ana-pantilie have worked on the start-up simplification in the past, and may have a better idea.

**Comment by virgil-serbanuta** [comment]
If I remember correctly, in the past there were some cases where we did something that looked like symbolic execution during normal execution, but I agree that this is almost surely related to the startup.

About startup simplification - I don't have a better idea. The only problem I can see is that this may behave unexpectedly if we set the "simplified" attribute without things being fully simplified.

For rewrite rules, the cases where this would actually matter should be very rare, since the LHS is almost always constructor-like.

We should be more careful with claims, though, which are simplified by the same code.
